repos/system_upgrade/el9toel10/actors/opensslenginescheck/libraries/opensslenginescheck.py

```diff
@@ -0,0 +1,154 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+
+
+# FIXME: This is taken from the el8toel9 library in
+# repos/system_upgrade/el8toel9/actors/opensslconfigcheck/libraries/opensslconfigcheck.py
+def _normalize_key(key):
+    """
+    Strip the part of the key before the first dot
+    """
+    s = key.split(".", 1)
+    if len(s) == 2:
+        return s[1]
+    return key
+
+
+def _key_equal(pair, key):
+    """
+    Check the keys are equal in OpenSSL configuration semantics
+
+    The OpenSSL semantics ignores everything before the first dot to allow specifying
+    something like following, where the first line would be otherwise normally ignored
+
+        TLS.MaxProtocol = TLSv1.3
+        DTLS.MaxProtocol = DTLSv1.2
+    """
+    if pair.key == key:
+        return True
+    return _normalize_key(pair.key) == key
+
+
+def _find_pair(block, key):
+    """
+    Find key-value pair in the given configuration block
+
+    In the given configuration block (OpenSslConfigBlock) find a key-value with a given key.
+    If multiple values match, only the last one is returned.
+    """
+    res = None
+    for pair in block.pairs:
+        if _key_equal(pair, key):
+            res = pair
+
+    return res
+
+
+def _openssl_find_block(config, name):
+    """
+    In the given configuration file (OpenSslConfig) find a block with a given name
+    """
+    for block in config.blocks:
+        if block.name == name:
+            return block
+
+    return None
+
+
+resources = [
+    reporting.RelatedResource('package', 'openssl'),
+    reporting.RelatedResource('file', '/etc/pki/tls/openssl.cnf')
+]
+
+
+def check_openssl_engines(config):
+    """
+    Check there are no engines configured in openssl.cnf
+
+    In case the of the pkcs11 engine, stop the upgrade as the package was removed from RHEL 10.
+
+    In case of other (usually third-party) engines, just warn about them being deprecated,
+    and suggest removal.
+    """
+    init_block = _openssl_find_block(config, "openssl_init")
```

Comment:
The name `openssl_init` can actually differ, too; what will not differ is that `openssl_conf` in the default (i.e., first) section will always contain the name of the init section.

Reply:
Good catch! We have the right value in `config.openssl_conf` so I will use that one.